File: 0001-two-sum/0001-two-sum.java
class Solution1 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int num = nums[i];
            int moreNeeded = target - num;

            if (map.containsKey(moreNeeded)) {
                return new int[]{map.get(moreNeeded), i};
            }

            map.put(num, i);
        }

        return new int[]{-1, -1};
    }
}

class Solution2{
    public int[] twoSum(int[] nums, int target){
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            int num = nums[i];
            int moreNeeded = target-num;
            if(map.containsKey(moreNeeded)){
                return new int[]{map.get(moreNeeded),i};
            }
            map.put(num,i);
        }
        return new int[]{-1,-1};
    }
}






class Solution{
    public int[] twoSum(int[] nums, int target){
        Map<Integer,Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            int num = nums[i];
            int moreNeeded = target-num;
            if(map.containsKey(moreNeeded)){
                return new int[]{map.get(moreNeeded),i};
            }
            map.put(num,i);
        }
        return new int[]{-1,-1};
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
// var twoSum = function(nums, target) {
//     let left = 0;
//     let right = nums.length - 1;

//     while (left < right) {
//         let sum = nums[left] + nums[right];

//         if (sum === target) {
//             return [left, right];
//         } else if (sum < target) {
//             left++;
//         } else {
//             right--;
//         }
//     }

//     return [];
// };



var twoSum = function(nums,target){
    const map = new Map();
    for(let i=0;i<nums.length;i++){
        const diff = target - nums[i];
        if(map.has(diff)){
            return [map.get(diff),i];
        }
        map.set(nums[i],i);
    }
    return [];
}